### DISCUSSION TITLE
What is the strategy with regard to 'the new Modern controls'?

### DISCUSSION THREAD

**Discussion by LenCardinal** [microsoft/powercat-creator-kit#314] [General]
Microsoft has released its first set of 'modern' controls for Canvas Apps. 

[Modern controls coming to canvas apps](https://powerapps.microsoft.com/en-us/blog/modern-controls-coming-to-canvas-apps/)

Where does this leave the Creator's Kit?  Creator's Kit appears richer at this point.  But reading the tea leaves...what should we assume about Creator's Kit futures?

Creator's Kit was updated monthly until last November.  The last update was the November release on December 5th.  Should we be reading into this?

**Comment by sperry1625** [comment]
Can anyone respond to these questions? I am investing a great deal of time leveraging these controls in all our application development efforts. Knowing the future is critical for proper planning and to create mitigation strategies.

Thank you!

**Comment by denise-msft** [comment]
Thanks for your questions and investment in using Creator Kit! 

The long-term guidance is to encourage users using Creator Kit to migrate to using the new modern controls where applicable, to ensure they are use the most robust components with the best support scenario. This is determined by:
* Whether the 'successor' of the control (such as Pivot) has been implemented in the modern controls suite (in this case as TabList), and 
* When appropriate (the modern controls are still in preview, therefore we do not recommend using in production until GA)

However, not all controls are scheduled to be implemented in modern controls. Our team intends to continue supporting the controls that are not available in the modern controls (as resources permit).

We will address this question in the office hours tomorrow in more detail and will release a more formal guidance shortly after (will post here). [Event link](https://msit.events.teams.microsoft.com/event/6bd357ad-55cb-4546-bf4e-dc0502ebd722@72f988bf-86f1-41af-91ab-2d7cd011db47)

**Comment by bcobra** [comment]
Hi, I am just wondering if there are any updates on this?

**Comment by denise-msft** [comment]
@sperry1625 @bcobra Please see my response for more details - is there any more information needed to help you determine your long term planning in relation to the Creator Kit?

We will continue to support the Creator Kit controls and it is here to stay. The simpler controls are being re-implemented as V9 components in the studio. The modern controls do not immediately have all of the theming and icon capabilities that Creator Kit has today but will be added over the next year. 

You can continue using the Creator Kit controls until the modern controls have the capabilities you need, and once the modern control version is at parity with the theming and icon capabilities of Creator Kit, we are planning on announcing a deprecation path. The first control we will likely try this out with is the elevation component, since the same capabilities (and more) are currently generally available in the layout container.

The more complex controls like SubwayNav that have clear value (based on usage and feedback) will continue to be supported in the Creator Kit.

As for new components, our team helps contribute to modern controls and your feedback helps us make prioritization decisions. We may add new components to the Creator Kit in the future, but more likely we will be shifting focus towards guidance and patterns/templates.

**Reply by bcobra** [reply]
Will the controls that are in a preview state ever be finished like the subway control?

**Reply by denise-msft** [reply]
Due to resource constraints, we don't have enough demand to prioritize the SubwayNav control into stable release in the foreseeable future, only providing bug fixes for highest priority areas based on overall usage.
